WalletWasabi.Backend/Controllers/BlockchainController.cs

```diff
@@ -204,12 +205,20 @@ private async Task<Transaction[]> FetchTransactionsAsync(uint256[] txIds, Cancel
 			{
 				// Ask to get missing transactions over RPC.
 				IEnumerable<Transaction> txs = await RpcClient.GetRawTransactionsAsync(txIdsRetrieve.Keys, cancellationToken).ConfigureAwait(false);
+
 				Dictionary<uint256, Transaction> rpcBatch = txs.ToDictionary(x => x.GetHash(), x => x);
 
 				foreach (KeyValuePair<uint256, Transaction> kvp in rpcBatch)
 				{
 					txIdsRetrieve[kvp.Key].TrySetResult(kvp.Value);
 				}
+
+				// RPC client does not throw if a transaction is missing, so we need to account for this case.
+				if (txs.Count() < txIdsRetrieve.Count)
+				{
+					IReadOnlyList<Exception> exceptions = MarkNotFinishedTasksAsFailed(txIdsRetrieve);
```

Comment:
We call `MarkNotFinishedTasksAsFailed` again in the `finally`. It won't harm because `tcs.Task.IsCompleted` but the execution path is strange. 
- What if you just return here and let finally handle the rest?